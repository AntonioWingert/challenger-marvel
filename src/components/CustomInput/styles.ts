import { styled } from "styled-components";
import { CustomInputProps } from ".";

export const Input = styled.input<CustomInputProps>`
  width: 50%;
  height: 4rem;
  border-radius: 10px;
  border: none;
  outline: none;
  text-align: left;
  padding-left: ${({ theme }) => theme.padding.lg};
  font-size: ${({ theme }) => theme["font-sizes"].sm};

  &::placeholder {
    color: #ccc;
    font-size: 1rem;
    padding-left: 1rem;
  }

  @media (max-width: 320px) {
    width: 80%;
    padding-left: 0;

    &::placeholder {
      padding-left: 0.4rem;
    }
  }
`;
