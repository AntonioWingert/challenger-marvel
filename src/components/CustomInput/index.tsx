import * as S from './styles';

export type CustomInputProps = {
  placeholder: string;
} & React.InputHTMLAttributes<HTMLInputElement>;

export default function CustomInput({ placeholder = 'Insira suas informações aqui', ...props}: CustomInputProps) {
  return (
    <>
      <S.Input placeholder={placeholder} {...props} />
    </>
  );
}