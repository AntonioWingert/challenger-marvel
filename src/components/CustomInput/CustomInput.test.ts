import { screen } from '@testing-library/react';
import render  from '../../utils/renderWithRouter';
import CustomInput from '.';


describe('<CustomInput />', () => {
  it('should render the input with all props passed', () => {
    const { container } = render(
      CustomInput({ placeholder: 'test' }),
    );
    
    expect(screen.getByPlaceholderText(/test/i)).toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });
});