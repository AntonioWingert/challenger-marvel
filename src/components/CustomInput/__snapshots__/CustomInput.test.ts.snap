// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CustomInput /> > should render the input with all props passed 1`] = `
.c0 {
  width: 50%;
  height: 4rem;
  border-radius: 10px;
  border: none;
  outline: none;
  text-align: left;
  padding-left: 16px;
  font-size: 1.6rem;
}

.c0::placeholder {
  color: #ccc;
  font-size: 1rem;
  padding-left: 1rem;
}

@media (max-width: 320px) {
  .c0 {
    width: 80%;
    padding-left: 0;
  }

  .c0::placeholder {
    padding-left: 0.4rem;
  }
}

<input
  class="c0"
  placeholder="test"
/>
`;
